dotnet-desktop-guide/framework/wpf/advanced/snippets/threading-model/csharp/MultiWindow.xaml.cs

```diff
@@ -0,0 +1,45 @@
+﻿//<full>
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace SDKSamples
+{
+    public partial class MultiWindow : Window
+    {
+        public MultiWindow() =>
+            InitializeComponent();
+
+        private void Window_Loaded(object sender, RoutedEventArgs e) =>
+            ThreadStatusItem.Content = $"Thread ID: {Thread.CurrentThread.ManagedThreadId}";
+
+        //<delay>
+        private void StartStopButton_Click(object sender, RoutedEventArgs e) =>
+            Task.Delay(TimeSpan.FromSeconds(5)).Wait();
```

Comment:
Hmmm, why wouldn't you do `async / await` here?

Reply:
This is demonstrating blocking the thread so you can see the windows that were launched under other threads continue to operate as normal.